### PR TITLE
Restart verdaccio server and registry path before e2e 

### PR DESCRIPTION
**Pull request by nikhname**
*Issue #, if available:*

*Description of changes:*
unset registry after verdaccio script, start and set again before e2e

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by yuth** [comment]
# [Codecov](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3433?src=pr&el=h1) Report
> Merging [#3433](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3433?src=pr&el=desc) into [master](https://codecov.io/gh/aws-amplify/amplify-cli/commit/97c893c34efab3758b1f2d21a47154e4fcb5bb82&el=desc) will **increase** coverage by `0.06%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3433?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master    #3433      +/-   ##
==========================================
+ Coverage   59.47%   59.54%   +0.06%     
==========================================
  Files         286      285       -1     
  Lines       13114    13099      -15     
  Branches     2493     2637     +144     
==========================================
  Hits         7800     7800              
+ Misses       4908     4876      -32     
- Partials      406      423      +17     
```


| [Impacted Files](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3433?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [packages/graphql-mapping-template/src/print.ts](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3433/diff?src=pr&el=tree#diff-cGFja2FnZXMvZ3JhcGhxbC1tYXBwaW5nLXRlbXBsYXRlL3NyYy9wcmludC50cw==) | `35.71% <0.00%> (ø)` | :arrow_up: |
| [...ages/amplify-appsync-simulator/src/server/index.ts](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3433/diff?src=pr&el=tree#diff-cGFja2FnZXMvYW1wbGlmeS1hcHBzeW5jLXNpbXVsYXRvci9zcmMvc2VydmVyL2luZGV4LnRz) | `64.28% <0.00%> (ø)` | :arrow_up: |
| [...ges/amplify-util-mock/src/CFNParser/stack/index.ts](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3433/diff?src=pr&el=tree#diff-cGFja2FnZXMvYW1wbGlmeS11dGlsLW1vY2svc3JjL0NGTlBhcnNlci9zdGFjay9pbmRleC50cw==) | `88.09% <0.00%> (ø)` | :arrow_up: |
| [...ges/graphql-transformer-core/src/util/fileUtils.ts](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3433/diff?src=pr&el=tree#diff-cGFja2FnZXMvZ3JhcGhxbC10cmFuc2Zvcm1lci1jb3JlL3NyYy91dGlsL2ZpbGVVdGlscy50cw==) | `11.76% <0.00%> (ø)` | :arrow_up: |
| [...es/graphql-transformer-core/src/stripDirectives.ts](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3433/diff?src=pr&el=tree#diff-cGFja2FnZXMvZ3JhcGhxbC10cmFuc2Zvcm1lci1jb3JlL3NyYy9zdHJpcERpcmVjdGl2ZXMudHM=) | `35.29% <0.00%> (ø)` | :arrow_up: |
| [...amplify-appsync-simulator/src/server/operations.ts](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3433/diff?src=pr&el=tree#diff-cGFja2FnZXMvYW1wbGlmeS1hcHBzeW5jLXNpbXVsYXRvci9zcmMvc2VydmVyL29wZXJhdGlvbnMudHM=) | `28.45% <0.00%> (ø)` | :arrow_up: |
| [...graphql-transformer-core/src/TransformerContext.ts](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3433/diff?src=pr&el=tree#diff-cGFja2FnZXMvZ3JhcGhxbC10cmFuc2Zvcm1lci1jb3JlL3NyYy9UcmFuc2Zvcm1lckNvbnRleHQudHM=) | `26.20% <0.00%> (ø)` | :arrow_up: |
| [...mplify-appsync-simulator/src/resolvers/function.ts](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3433/diff?src=pr&el=tree#diff-cGFja2FnZXMvYW1wbGlmeS1hcHBzeW5jLXNpbXVsYXRvci9zcmMvcmVzb2x2ZXJzL2Z1bmN0aW9uLnRz) | `9.67% <0.00%> (ø)` | :arrow_up: |
| [...plify-appsync-simulator/src/server/subscription.ts](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3433/diff?src=pr&el=tree#diff-cGFja2FnZXMvYW1wbGlmeS1hcHBzeW5jLXNpbXVsYXRvci9zcmMvc2VydmVyL3N1YnNjcmlwdGlvbi50cw==) | `26.81% <0.00%> (ø)` | :arrow_up: |
| [...lify-graphql-types-generator/src/compiler/index.ts](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3433/diff?src=pr&el=tree#diff-cGFja2FnZXMvYW1wbGlmeS1ncmFwaHFsLXR5cGVzLWdlbmVyYXRvci9zcmMvY29tcGlsZXIvaW5kZXgudHM=) | `90.32% <0.00%> (ø)` | :arrow_up: |
| ... and [20 more](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3433/diff?src=pr&el=tree-more) | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3433?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3433?src=pr&el=footer). Last update [97c893c...aad763f](https://codecov.io/gh/aws-amplify/amplify-cli/pull/3433?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: .circleci/local_publish_helpers.sh [hunk 1]
```diff
@@ -13,15 +13,11 @@ function startLocalRegistry {
...
-  # Set registry to local registry
-  npm set registry "$custom_registry_url"
-  yarn config set registry "$custom_registry_url"
-
   # Login so we can publish packages
   (cd && npx npm-auth-to-token@1.0.0 -u user -p password -e user@example.com -r "$custom_registry_url")
 }
 
-function stopLocalRegistry {
+function unsetNpmRegistryPath {
```
Comment (yuth) [inline .circleci/local_publish_helpers.sh]: this should be called unsetNpmRegistryUrl

File: .circleci/local_publish_helpers.sh [hunk 2]
```diff
@@ -31,4 +27,10 @@ function changeNpmGlobalPath {
   mkdir ~/.npm-global
   npm config set prefix '~/.npm-global'
   export PATH=~/.npm-global/bin:$PATH
+}
+
+function setNpmRegistryPathToLocal {
```
Comment (yuth) [inline .circleci/local_publish_helpers.sh]: change `setNpmRegistryPathToLocal` to `setNpmRegistryUrlToLocal`

**Comment by yuth** [comment]
Could you rebase this with master. We have updated circle ci workflow to fix failing e2e and mock_e2e tests

**Comment by nikhname** [comment]
> Could you rebase this with master. We have updated circle ci workflow to fix failing e2e and mock_e2e tests

Rebased

File: .circleci/config.yml [hunk 3]
```diff
@@ -110,12 +119,20 @@ jobs:
           command: |
             source .circleci/local_publish_helpers.sh
             startLocalRegistry "$(pwd)/.circleci/verdaccio.yaml"
+            setNpmRegistryUrlToLocal
             changeNpmGlobalPath
+            setNpmRegistryUrlToLocal
```
Comment (yuth) [inline .circleci/config.yml]: why repeat?

File: package.json [hunk 4]
```diff
@@ -55,34 +55,33 @@
     "@commitlint/cli": "^8.1.0",
     "@commitlint/config-conventional": "^8.1.0",
     "@commitlint/config-lerna-scopes": "^8.1.0",
-    "@typescript-eslint/parser": "^2.16.0",
+    "@types/jest": "^25.1.1",
```
Comment (yuth) [inline package.json]: why there are changes here?

**Pull request merged**